typings/common.d.ts

```diff
@@ -9,7 +9,14 @@ declare global {
 		[key: string]: T;
 	}
 
+	type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
 	type Writable<T> = { -readonly [K in keyof T]: T[K] };
+	type NonNullableField<T, F extends keyof T> = Overwrite<
+		T,
+		{
+			[P in F]: NonNullable<T[P]>;
+		}
+	>;
```

Comment:
That's not the same though, since `Required` only removes the optionality of the property (`-?`), while `NonNullable` also removes `null | undefined` from the prop value, which is what I needed it for.
I think we had a similar discussion in the cloud api as well:
See: https://github.com/balena-io/balena-api/pull/2856#discussion_r532805169

Reply:
You're right. It'd be good to export these types to reuse in cloud api though rather than copy/pasting, like the pinejs types are exposed

Reply:
I agree :+1: will do in a follow-up PR. The only bad thing would be that we would probably need to move the common-types under src and import them explicitly in all files. But we might be able to avoid the second point, by having adding `src/common-types.ts` in the tsconfig's `include` section :thinking: